streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java

```diff
@@ -329,10 +329,11 @@ public void testAddStateStoreWithSink() {
     }
 
     @Test
-    public void testAddStateStoreWithDuplicates() {
+    public void testAddStateStoreWithDifferentInstances() {
```

Comment:
Can we improve the test name -> `shouldNotAllowToAddStoresWithSameName` (or similar)

We should also add a test that verifies that adding the same `StoreBuilder` instance multiple times works.